src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,193 @@
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexForGet(index);
+        Node<T> currentNode = getNodeByIndex(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> currentNode = getNodeByIndex(index);
+        removeNode(currentNode);
+        return currentNode.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> currentNode = getNodeByValue(object);
+        return removeNode(currentNode);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addToFirstPosition(T value) {
+        Node<T> newNode = new Node<>(null, value, head);
+        if (head != null) {
+            head.prev = newNode;
+        } else {
+            tail = newNode;
+        }
+        head = newNode;
+        size++;
+    }
+
+    private void addToLastPosition(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail != null) {
+            tail.next = newNode;
+        } else {
+            head = newNode;
+        }
+        tail = newNode;
+        size++;
+    }
+
+    private void addInsideList(int index, T value) {
+        Node<T> currentNode = getNodeByIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        addNodeInsideList(newNode, currentNode);
+    }
+
+    private void addNodeInsideList(Node<T> newNode, Node<T> oldNode) {
+        if (oldNode == head) {
+            addToFirstPosition(newNode.value);
+        } else {
+            newNode.prev = oldNode.prev;
+            oldNode.prev.next = newNode;
+            oldNode.prev = newNode;
+            newNode.next = oldNode;
+        }
+        size++;
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || (size == 0 ? index > 0 : index > size)) {
+            throw new IndexOutOfBoundsException("incorrect index");
+        }
+    }
+
+    private void checkIndexForGet(int index) {
+        if (index < 0 || (size == 0 ? index > 0 : index >= size)) {
+            throw new IndexOutOfBoundsException("incorrect index");
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndexForGet(index);
+        Node<T> currentNode = null;
+        if (index <= size / 2) {
+            int count = 0;
+            currentNode = head;
+            while (count != index) {
+                if (index == count++) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        } else {
+            int count = size - NUMBER_ONE;
```

Comment:
NUMBER_ONE it's not quite informative constant. 
In this case, you can just use int count = size - 1;